### PR TITLE
Use provided uuid as primary key

### PR DESCRIPTION
**Pull request by cprieto**
- Fix https://github.com/prisma/prisma/issues/12528
 - Now if uuid is provided we use that one and not generate a new uuid

### CODE REVIEW

File: query-engine/connectors/sql-query-connector/src/database/operations/write.rs [hunk 1]
```diff
@@ -20,12 +20,13 @@ async fn generate_id(
...
     primary_key: &FieldSelection,
     trace_id: Option<String>,
+    args: &WriteArgs,
 ) -> crate::Result<Option<SelectionResult>> {
     // Go through all the values and generate a select statement with the correct MySQL function
     let (pk_select, need_select) = primary_key
         .selections()
         .filter_map(|field| match field {
-            SelectedField::Scalar(x) if x.default_value.is_some() => x
+            SelectedField::Scalar(x) if x.default_value.is_some() && !args.has_arg_for(x.name()) => x
```
Comment (dpetrick) [inline query-engine/connectors/sql-query-connector/src/database/operations/write.rs:29]: `WriteArgs` contain the data source field name, which is not the `x.name()`, but the `x.db_name()`. Fairly sure that if you add a `@map` to the uuid field your test will fail.

**Review by dpetrick — CHANGES_REQUESTED** [review]

File: query-engine/connectors/sql-query-connector/src/database/operations/write.rs (hunk 1 above)
Comment (cprieto) [reply query-engine/connectors/sql-query-connector/src/database/operations/write.rs:29]: Fair enough

**Review by dpetrick — APPROVED** [review]

**Comment by floriantz** [comment]
Thanks a lot ❤️

**Comment by FredericEspiau** [comment]
@floriantz told you this was the fix 😄

**Pull request merged**